kuba-kostrzewski/simpleToDoApp/src/main/java/pl/kostrzej/simpleToDoApp/components/validator/UserValidator.java

```diff
@@ -0,0 +1,39 @@
+package pl.kostrzej.simpleToDoApp.components.validator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.kostrzej.simpleToDoApp.components.user.User;
+import pl.kostrzej.simpleToDoApp.components.user.UserRepository;
+
+@Component
+@Slf4j
+public class UserValidator {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    public void validate(User user){
+        log.info("User validation process initialized.");
+        if (user.getLogin().isEmpty()) {
+            log.info("Login is empty.");
+            throw new NullPointerException("Login nie może być pusty!");
```

Comment:
Nie jest dobrmy pomysłem rzucanie NPE - jest to rzadko widywany wyjątek i nie ma co straszyć innych programistów ich wystąpieniem. Lepiej wykorzystać stworzone przez siebie wyjątki,.

Reply:
poprawione na `IllegalArgumentException`